lib/compress/zstd_compress.c

```diff
@@ -465,14 +467,17 @@ size_t ZSTD_CCtxParam_setParameter(
  * The multithreading parameters jobSize and overlapSizeLog are set only if
  * nbThreads > 1.
  *
- * Pledged srcSize is treated as unknown.
+ * pledgedSrcSize is considered unknown
  */
 size_t ZSTD_CCtx_setParametersUsingCCtxParams(
         ZSTD_CCtx* cctx, const ZSTD_CCtx_params* params)
 {
     if (cctx->streamStage != zcss_init) return ERROR(stage_wrong);
     if (cctx->cdict) return ERROR(stage_wrong);
 
+#if 1
```

Comment:
Is this intentional, or accidentally left in from debugging?

Reply:
Indeed, I just changed this behavior, and kept the old code in, in case the new path would trigger some error in future tests (most likely in `zstreamtest --opaqueapi`).

I've seen no error so far, but was willing to wait a bit longer before removing the old path.

Reply:
I remember there being a subtle reason why it is there, but I don't remember exactly what it is. I'll take a deeper look.

Reply:
Not setting `loadedDictEnd` would nullify the effect of `ZSTD_p_forceMaxWindow` right?

Reply:
To be checked

Reply:
I believe setting `loadedDictEnd=0` has no practical impact.

The impact of `forceWindow` happens during dictionary loading, as it should : 
https://github.com/facebook/zstd/blob/dev/lib/compress/zstd_compress.c#L1890
There, `loadedDictEnd` is set unconditionally, irrespective of its previous value.

Just to be on the safe side, since the main user of `forceWindow` is ZSTDMT, I tested it at different compression settings : compression result is strictly identical, at all levels, wether `loadedDictEnd` is set to `0` or not. Also ran fuzzer tests, no impact either.

As a consequence, changing `cctx->loadedDictEnd` will be removed. 
With this change, setting a compression parameter doesn't produce any side effect.